assets/javascripts/initializers/code-bytes.js

```diff
@@ -135,15 +135,19 @@ function initializeCodeByte(api) {
     },
   });
 
-  function renderCodebyteFrame(language = '', text = '') {
+  function renderCodebyteFrame(language = '', text = '', isPreview = false) {
     return loadScript(
       'https://cdn.jsdelivr.net/npm/js-base64@3.6.0/base64.min.js'
     ).then(() => {
       const frame = document.createElement('iframe');
-
-      const encodedURI = Base64.encodeURI(text);
       frame.allow = 'clipboard-write';
-      frame.src = `https://www.codecademy.com/codebyte-editor?lang=${language}&text=${encodedURI}`;
+
+      const params = [];
+      params.push(`lang=${language}`);
+      params.push(`text=${Base64.encodeURI(text)}`);
+      params.push(`mode=${isPreview ? 'compose' : 'view'}`);
```

Comment:
We'll want to expand or tweak this. We could either add a second parameter telling the iframe that we are in Discourse (as opposed to CC Docs), or just have a single parameter that encapsulates both pieces of information (e.g. it could have values 'ForumsComposer', 'ForumsView', and 'Docs').

Reply:
+1 to the enum strategy!

Reply:
sorry, to be specific I think it's a good idea to encompass all this info in one parameter rather than juggling 2. It looks like we will need at least the ones Doug mentioned right now.

Reply:
Combining this would probably be problematic, as it would multiply the possible value and the checks we would have to make (for example if (mode === ForumsView || mode === Docs)).  That being said, I think the referrer of the parent page will give us all the information we need about the site/page this is embedded on.